nil/services/rpc/rawapi/shard_api.go

```diff
@@ -0,0 +1,76 @@
+package rawapi
+
+import (
+	"context"
+
+	"github.com/NilFoundation/nil/nil/common"
+	"github.com/NilFoundation/nil/nil/common/logging"
+	"github.com/NilFoundation/nil/nil/common/sszx"
+	"github.com/NilFoundation/nil/nil/internal/network"
+	"github.com/NilFoundation/nil/nil/internal/types"
+	rawapitypes "github.com/NilFoundation/nil/nil/services/rpc/rawapi/types"
+	rpctypes "github.com/NilFoundation/nil/nil/services/rpc/types"
+	"github.com/NilFoundation/nil/nil/services/txnpool"
+)
+
+type shardApiBase interface {
+	shardId() types.ShardId
+	setNodeApi(nodeApi NodeApi)
+	setAsP2pRequestHandlersIfAllowed(ctx context.Context, networkManager network.Manager, logger logging.Logger) error
+}
+
+const apiNameRo = "rawapi_ro"
```

Comment:
Does it mean that we will call methods like `/shard/5/rawapi_ro/GetBlockHeader` instead of `/shard/5/rawapi/GetBlockHeader`?

Reply:
Yes

Reply:
Do we really need such inconsistency?

Reply:
I think this is worth doing. First, since now the interfaces exported to libp2p are not linked, we might accidentally get an unwanted name collision. Second, we can imagine a situation where we might want to have a collision of method names in different interfaces. For example, the same `GetTransactionCount` could have a readonly version without `PendingBlock` support.